scripts/ci-entrypoint.sh

```diff
@@ -94,14 +95,20 @@ setup() {
         echo ''
     )}"
     export AZURE_RESOURCE_GROUP="${CLUSTER_NAME}"
-    export AZURE_LOCATION="${AZURE_LOCATION:-$(capz::util::get_random_region)}"
-    echo "Using AZURE_LOCATION: ${AZURE_LOCATION}"
+    if [ "${WORKER_MACHINE_COUNT}" -gt "10" ]; then
+        export AZURE_LOCATION="${AZURE_LOCATION:-$(capz::util::get_random_region_load)}"
+        echo "Using AZURE_LOCATION: ${AZURE_LOCATION}"
+    else
+        export AZURE_LOCATION="${AZURE_LOCATION:-$(capz::util::get_random_region)}"
+        echo "Using AZURE_LOCATION: ${AZURE_LOCATION}"
+    fi
+    # TODO these AZURE_LOCATION_* overrides may have the effect of
+    # disassociating VM regions from disks, leading to attachment failures.
+    # Less likely with GPU scenarios but FYI.
```

Comment:
This does appear to be the case. All ci-entrypoint runs use a custom cloud-provider config aside from the one CAPZ normally generates based on a cluster: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/8e98848bccebd4df3c826c8e378e3e688ccaa8ba/hack/create-custom-cloud-provider-config.sh

That references a file in the cloud-provider-azure repo that depends on the `AZURE_LOCATION` environment variable.